Software/Development/SemesterProject/App/analogAcquisitionManager.c

```diff
@@ -0,0 +1,439 @@
+//
+//
+//###########################################################################
+
+//###########################################################################
+
+//
+// Includes
+//
+#include "analogAcquisitionManager.h"
+#include "DSP28x_Project.h"
+
+
+//
+// Quasi-global variables definition
+//
+
+/*
+ * Two different structs are created for the signals. One with a high priority (The currents),
+ * and one with a low priority (The rest). The reason for this is to keep the execution time for
+ * reading the currents as low as possible, such that we are able to read the values within the
+ * limits of our switching frequency.
+ */
+static struct
+{
+    AnalogSignal currentMeasA;
+    AnalogSignal currentMeasB;
+    AnalogSignal currentMeasC;
+} CurrentSignalList;
+
+static struct
+{
+    AnalogSignal voltageMeas24;
+    AnalogSignal voltageMeas36;
+    AnalogSignal thermalMeas1;
+    AnalogSignal thermalMeas2;
+    AnalogSignal sliderPotLeft;
+    AnalogSignal sliderPotRight;
+    AnalogSignal rotaryPot1;
+    AnalogSignal rotaryPot2;
+    AnalogSignal rotaryPot3;
+    AnalogSignal connectorPot1;
+    AnalogSignal connectorPot2;
+} AnalogSignalList;
+
+
+/*
+ * Reads and filters the values of signals in the high priority struct.
+ * This function will be called by the adc_isr() every time the timer1 trigger.
+ */
+void readHighPrioritySignals(void)
+{
+    readAnalogSignals(&CurrentSignalList, sizeof(CurrentSignalList));
+    calculateFilteredValue(&CurrentSignalList, sizeof(CurrentSignalList));
+}
+
+/*
+ * Reads and filters the values of signals in the low priority struct.
+ * This function will be called by the scheduler when ever.
+ */
+void readLowPrioritySignals(void)
+{
+    readAnalogSignals(&AnalogSignalList, sizeof(AnalogSignalList));
+    calculateFilteredValue(&AnalogSignalList, sizeof(AnalogSignalList));
+}
+
+/*
+ * This takes the address and the size of the struct of interest, and iterates through it to
+ * read the ADC value from the register
+ */
+void readAnalogSignals(void *signal, int size)
+{
+    AnalogSignal *structPointer;
+    AnalogSignal *initialMemoryPosition = signal;
+    AnalogSignal *finalMemoryPosition = initialMemoryPosition + size/sizeof(AnalogSignal);
+
+    for (structPointer = initialMemoryPosition; structPointer < finalMemoryPosition; structPointer++)
+        readADCValue(structPointer);
+}
+
+/*
+ * This takes the address and the size of the struct of interest, and iterates through it to
+ * calculate the filtered voltage measured
+ */
+void calculateFilteredValue(void *signal, int size)
+{
+    AnalogSignal *structPointer;
+    AnalogSignal *initialMemoryPosition = signal;
+    AnalogSignal *finalMemoryPosition = initialMemoryPosition + size/sizeof(AnalogSignal);
+
+    for (structPointer = initialMemoryPosition; structPointer < finalMemoryPosition; structPointer++)
+        filterADCValue(structPointer);
+}
+
+Uint16 getAnalogErrorStatus(void)

```

Comment:
It would be nice to add a comment about what this function is doing and what is it's output.